Client/mods/deathmatch/logic/lua/CLuaFunctionDefs.World.cpp

```diff
@@ -1920,3 +1922,147 @@ int CLuaFunctionDefs::FetchRemote(lua_State* luaVM)
     lua_pushboolean(luaVM, false);
     return 1;
 }
+
+// table getRemoteRequests([resource theResource = nil])
+int CLuaFunctionDefs::GetRemoteRequests(lua_State* luaVM)
+{
+    CScriptArgReader argStream(luaVM);
+    CResource*       pResource = nullptr;
+    CLuaMain*        pLuaMain = nullptr;
+    int              iIndex = 0;
+
+    argStream.ReadUserData(pResource, NULL);
+
+    // Grab virtual machine
+    if (pResource)
+        pLuaMain = pResource->GetVM();
+
+    lua_newtable(luaVM);
+    for (const auto& request : g_pClientGame->GetRemoteCalls()->GetCalls())
+    {
+        if(!pLuaMain || request->GetVM() == pLuaMain)
+        {
+            lua_pushnumber(luaVM, ++iIndex);
+            lua_pushuserdata(luaVM, request);
+            lua_settable(luaVM, -3);
+        }
+    }
+
+    return 1;
+}
+
+// table getRemoteRequestInfo(element theRequest[, number postDataLength = 0])
+int CLuaFunctionDefs::GetRemoteRequestInfo(lua_State* luaVM)
+{
+    CScriptArgReader argStream(luaVM);
+    CLuaArguments    info, requestedHeaders;
+    CRemoteCall*     pRemoteCall = nullptr;
+    CResource*       pThisResource = g_pClientGame->GetResourceManager()->GetResourceFromLuaState(luaVM);
+    int              iPostDataLength = 0;
+
+    argStream.ReadUserData(pRemoteCall);
+    argStream.ReadNumber(iPostDataLength, 0);
+
+    if (!argStream.HasErrors())
+    {
+        CResource* pResource = pRemoteCall->GetVM()->GetResource();
+        bool bExtendedInfo = (pResource == pThisResource);
+        
+        info.PushString("type");
+        info.PushString((pRemoteCall->IsFetch() ? "fetch" : "call"));
+
+        // remove query_string from url when bExtendedInfo isn't set
+        SString sURL = pRemoteCall->GetURL();
+
+        if (!bExtendedInfo)
+            sURL = sURL.ReplaceI("%3F", "?").Replace("#", "?").SplitLeft("?");
+
+        info.PushString("url");
+        info.PushString(sURL);
+
+        info.PushString("queue");
+        info.PushString(pRemoteCall->GetQueueName().c_str());
+        
+        info.PushString("resource");
+        
+        if (pResource)
+            info.PushResource(pResource);
+        else
+            info.PushBoolean(false);
+
+        info.PushString("start");
+        info.PushNumber(pRemoteCall->GetStartTime());
+
+        if (bExtendedInfo)
+        {
+            SString sPostData = pRemoteCall->GetOptions().strPostData.c_str();
+
+            if (iPostDataLength > 0)
+                sPostData = sPostData.SubStr(0, iPostDataLength);
```

Comment:
If `iPostDataLength` is 0, then all the post data is included. Is that the intention?

Reply:
Yes, it was my intentation

Reply:
The problem I have with including postdata by default (and headers to a lesser extent) is that
they take memory and cpu, don't change during the download and might not be required by scripters

Reply:
How about defaulting `iPostDataLength` to something like 100 ? Or shall we remove it entirely ?

Reply:
Defaulting to 100 is weird and will cause issues with scripters only testing things for low amounts of post data. I think a better option would be to expect an option table to include the more resource intensive data on-demand only. 

`getRemoteRequestInfo(..., { postDataLength = 0 })` -> Infinite length
`getRemoteRequestInfo(...)` -> No post data
`getRemoteRequestInfo(..., { postDataLength = 500 })` -> Limited to 500.

Same thing with headers:

`getRemoteRequestInfo(..., { includeHeaders = true })` -> Adds headers

Reply:
Good idea, lemme do that one aswell. What options should be there besides headers and postData ?